docs/concepts.rst

```diff
@@ -1,11 +1,86 @@
+
+CumulusCI knows which project you are working on by the current working directory of your shell. Make sure to change directories inside your project before you run project-specific commands.
+
+Managing Orgs & Services with the Keychain
+------------------------------------------
+
+CumulusCI gives each project a keychain, and also offers a global keychain that's shared across projects. The keychain's role is to store access information for all of the orgs you're using with the project — both scratch orgs and persistent orgs — and the details of the services you've connected, such as a GitHub account or an instance of MetaDeploy.
+
+Services
+++++++++
+
+Services are usually, but not always connected to CumulusCI across projects: they live in the global keychain. A service represents functionality external to CumulusCI that you authenticate with in order to achieve a workflow. For example, a GitHub account or MetaDeploy account would be represented by a service.
+
+The command ``cci service list`` shows you which services are connected in the context of the current project.
+
+Some services can be connected at the project level, which means that they're scoped to a single project and aren't shared. We recommend primarily using global services because they're easier to use, but you may encounter scenarios where, for example, you need to use a specific Dev Hub for one and only one project. Connecting that service with ``cci service connect devhub --project`` supports that use case.
+
+CumulusCI stores service authentication details in an encrypted file in your configuration directory.
+
+Orgs
+++++
+
+Each project's keychain stores authentication information for all of the orgs that are in use by the project, including scratch orgs, persistent orgs like a production or packaging org, and information about scratch orgs that are yet to be created.
+
+The ``cci org list`` command shows all of the connected orgs in the project keychain, as well as defined scratch org configurations that have not yet been built. (TODO: reference org configuration section).
+
+When CumulusCI builds a scratch org, it automatically shares the org with your Salesforce DX keychain, but names the org in a way that helps keep orgs separate between projects. For example, if you build a ``dev`` org in the project ``Test``, CumulusCI will call that org ``dev`` in your CumulusCI keychain, and ``Test__dev`` in the Salesforce DX keychain. This prevents your scratch orgs from colliding across projects.
+
+When you attach a persistent org to a project's keychain using ``cci org connect``, that org is not added to the Salesforce DX keychain - it belongs to the project alone.
+
+CumulusCI stores org authentication details in an encrypted file in your configuration directory.
```

Comment:
I'd probably leave the last 3 paras out of the key concepts doc. The key concept is that each project has its own set of orgs which can include persistent and scratch orgs.